ee/app/livechat-enterprise/server/hooks/checkDefaultAgentOnNewRoom.js

```diff
@@ -0,0 +1,36 @@
+import { callbacks } from '../../../../../app/callbacks';
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { settings } from '../../../../../app/settings';
+import { LivechatRooms, LivechatVisitors, Users } from '../../../../../app/models/server';
+
+callbacks.add('livechat.checkDefaultAgentOnNewRoom', (defaultAgent, defaultGuest) => {
+	if (defaultAgent || !defaultGuest) {
+		return defaultAgent;
+	}
+
+	if (!RoutingManager.getConfig().autoAssignAgent) {
+		return defaultAgent;
+	}
+
+	if (!settings.get('Livechat_last_chatted_agent_routing')) {
+		return defaultAgent;
+	}
+
+	const { _id: guestId } = defaultGuest;
+	const guest = LivechatVisitors.findOneById(guestId, { fields: { lastAgent: 1 } });
+	const { lastAgent: { username: usernameByVisitor } = {}, token } = guest;
```

Comment:
you are not requesting the `token` field from the query above. this will not work